src/vs/platform/terminal/node/ptyHostService.ts

```diff
@@ -191,7 +163,13 @@ export class PtyHostService extends Disposable implements IPtyService {
 		this._register(proxy.onProcessOrphanQuestion(e => this._onProcessOrphanQuestion.fire(e)));
 		this._register(proxy.onDidRequestDetach(e => this._onDidRequestDetach.fire(e)));
 
-		return [client, proxy];
+		// HACK: When RemoteLoggerChannelClient is not delayed, the Pty Host log file won't show up
+		// in the Output view of the first window?
+		Event.once(Event.any(proxy.onProcessReady, proxy.onProcessReplay))(() => {
+			this._register(new RemoteLoggerChannelClient(this._loggerService, client.getChannel(TerminalIpcChannels.Logger)));
+		});
```

Comment:
Not sure why but logging will not work without delaying it like this, it seems to register the logger when called regardless, it just doesn't show up in the first window's output view 😕